### PR TITLE
EVG-19914 consolidate patch/version statuses

### PR DESCRIPTION
**Pull request by ablack12**
EVG-19914
### Description
Remove patch statuses in favor of version statuses (except for succeeded statuses, where we'll keep the patch status for both patches and versions, since this is the UI expected status). Created EVG-20032 to do a final cleanup of this in a year (since old versions may still have the "success" status). Removed some unneeded helper functions along the way.

### Testing
Verified existing tests pass; will do some staging things to also verify.

### Documentation
Updated a patch trigger alias to clarify that we use "succeeded" and not "success" (this was already true).

### CODE REVIEW

File: trigger/patch.go [hunk 1]
```diff
@@ -93,13 +95,13 @@ func (t *patchTriggers) patchOutcome(sub *event.Subscription) (*notification.Not
...
-		if ps != evergreen.PatchSucceeded && ps != evergreen.PatchFailed && ps != evergreen.PatchAllOutcomes {
+		if !evergreen.IsFinishedVersionStatus(ps) && ps != patchAllOutcomes {
 			return nil, nil
 		}
 
-		successOutcome := (ps == evergreen.PatchSucceeded) && (t.data.Status == evergreen.PatchSucceeded)
-		failureOutcome := (ps == evergreen.PatchFailed) && (t.data.Status == evergreen.PatchFailed)
-		anyOutcome := ps == evergreen.PatchAllOutcomes
+		successOutcome := evergreen.IsSuccessfulVersionStatus(ps) && evergreen.IsSuccessfulVersionStatus(t.data.Status)
+		failureOutcome := (ps == evergreen.VersionFailed) && (t.data.Status == evergreen.VersionFailed)
```
Comment (Kimchelly) [inline trigger/patch.go:103]: nit: remove unnecessary parentheses since the order of operations precedence will evaluate it the same anyways.

File: trigger/version.go [hunk 2]
```diff
@@ -120,17 +120,13 @@ func (t *versionTriggers) makeData(sub *event.Subscription, pastTenseOverride st
 		}
 
 		// Look at collective status because we don't know whether the last patch to finish in the version was a child or a parent.
-		patchStatus, err := p.CollectiveStatus()
+		versionStatus, err = p.CollectiveStatus()
```
Comment (Kimchelly) [inline trigger/version.go:123]: nit: I would still call this `patchStatus` since it's for the patch document and not the version.

**Review by Kimchelly — CHANGES_REQUESTED** [review]
I would add a doc comment to the `(Patch).Status` field that it shares the same statuses with versions.

File: trigger/version.go (hunk 2 above)
Comment (ablack12) [reply trigger/version.go:123]: Personally I'd rather not, since this is being used for a version trigger.

File: globals.go [hunk 3]
```diff
@@ -131,20 +131,17 @@ const (
...
-
-	// VersionAborted and PatchAborted are display statuses only and not stored in the DB
+	VersionStarted = "started"
+	VersionCreated = "created"
+	VersionFailed  = "failed"
+
+	// LegacyVersionSucceeded should be removed in EVG-20032. In the meantime,
+	// will set with the new status but consider both statuses.
+	LegacyVersionSucceeded = "success"
+	VersionSucceeded       = "succeeded"
```
[the remark this answers is not in the document]
Comment (khelif96) [reply globals.go]: We would need to make some changes to the UI but they shouldn't be too heavy. 
https://github.com/evergreen-ci/evergreen/blob/8102969ec79bea081e18d5883c537c37e8248d48/graphql/user_resolver.go#L28 Would likely need to be updated to support backward compatibility between filtering on the new and old statuses.

And we would need to update the references on the UI to support the new status naming as well (While maintaining backward compatibility) 

So TLDR 2 smallish tickets

[the remark this answers is not in the document]
Comment (Kimchelly) [reply globals.go]: I'm still curious about which one we're going to use ("succeeded" vs "success"), but the rest of the changes LGTM.

[the remark this answers is not in the document]
Comment (ablack12) [reply globals.go]: Yeahhh I hear what you're saying about consistency. I'll update this ticket to update the UI filter (and verify there aren't any others) and made EVG-20065 to update the UI to understand both statuses.

[the remark this answers is not in the document]
Comment (ablack12) [reply globals.go]: Feel free to re-review!

[the remark this answers is not in the document]
Comment (khelif96) [reply globals.go]: Updated logic makes sene to me. Don't forget to deploy the UI changes before this is deployed.

File: docs/API/REST-V2-Usage.md [hunk 4]
```diff
@@ -695,7 +695,7 @@ A patch is a manually initiated version submitted to test local changes.
 | git_hash              | string         | Hash of commit off which the patch was initiated                                                                                     |
 | patch_number          | int            | Incrementing counter of user's patches                                                                                               |
 | author                | string         | Author of the patch                                                                                                                  |
-| status                | string         | Status of patch                                                                                                                      |
+| status                | string         | Status of patch (created, started, success, failed)                                                                                  |
```
Comment (Kimchelly) [inline docs/API/REST-V2-Usage.md:698]: I just documented a bunch of these in #6580, so maybe remove so we can avoid merge conflicts?

**Review by Kimchelly — APPROVED** [review]
LGTM mod one merge conflict suggestion. Thanks for consolidating these 🙏

**Comment by ablack12** [comment]
I'm going to close this and revisit to make this more backwards compatible with the UI.
